### ISSUE TITLE
Does not detect type incompatibility with tuple type

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7115]
The following code should result in a type violation, but it does not currently.

```python
def func2(t3: tuple[int, ...]):
    v2: tuple[int, *tuple[int, ...]] = t3
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.349, which I just published.
